src/core/CorePool.ts

```diff
@@ -325,12 +409,56 @@ export class CorePool {
     tickUpper: number,
     liquidityDelta: JSBI
   ): Position {
-    // TODO
-    return new Position();
+    let _position: Position = this.getPosition(owner, tickLower, tickUpper);
+
+    let _tick = this.tickManager.getTickAndInitIfAbsent(this._tickCurrent);
+
+    let flippedLower = _tick.update(
+      liquidityDelta,
+      this._tickCurrent,
+      this._feeGrowthGlobal0X128,
+      this._feeGrowthGlobal1X128,
+      false,
+      this.maxLiquidityPerTick
+    );
+
+    let flippedUpper = _tick.update(
+      liquidityDelta,
+      this._tickCurrent,
+      this._feeGrowthGlobal0X128,
+      this._feeGrowthGlobal1X128,
+      true,
+      this.maxLiquidityPerTick
+    );
+
+    let _feeGrowthInsideStep = this.tickManager.getFeeGrowthInside(
+      tickLower,
+      tickUpper,
+      this._tickCurrent,
```

Comment:
Nit: If we want to read value from a property of class, let's use property getter, so yeah _ is not needed.

Reply:
Ok, it's good point to use getter func, make code much more clear!